packages/amplify-codegen/tests/commands/models.test.js

```diff
@@ -71,42 +71,128 @@ describe('command - models', () => {
             expect(fs.readdirSync(outputDirectory).length).toBeGreaterThan(0);
         });
+
+        it(frontend + ': Should clear the output directory before generation if cleanGeneratedModelsDirectory is true', async () => {
+            // mock the input and output file structure
+            const schemaFilePath = path.join(MOCK_BACKEND_DIRECTORY, 'api', MOCK_PROJECT_NAME);
+            const outputDirectory = path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend]);
+            const preExistingFileName = 'preexisting.txt';
+            const preExistingFilePath = path.join(outputDirectory, preExistingFileName);
+            const mockedFiles = {};
+            mockedFiles[schemaFilePath] = {
+                'schema.graphql': ' type SimpleModel { id: ID! status: String } '
+            };
+            mockedFiles[outputDirectory] = {};
+            mockedFiles[outputDirectory][preExistingFileName] = 'A pre-existing file to be cleared';
+            mock(mockedFiles);
+            MOCK_CONTEXT.amplify.getProjectConfig.mockReturnValue({frontend: frontend});
+
+            require('amplify-cli-core').FeatureFlags.getBoolean.mockImplementation((name, defaultValue) => {
+                if (name === 'codegen.useappsyncmodelgenplugin' ||
+                    name === 'codegen.cleanGeneratedModelsDirectory') {
+                  return true;
+                }
+            });
+
+            // assert output directory has a mock file to be cleared
+            expect(fs.readdirSync(outputDirectory).length).toEqual(1);
+            expect(fs.existsSync(preExistingFilePath));
+
+            await generateModels(MOCK_CONTEXT);
+
+            // assert model generation succeeds
+            expect(graphqlCodegen.codegen).toBeCalled();
+
+            // assert that codegen generated in correct place
+            expect(fs.readdirSync(outputDirectory).length).toBeGreaterThan(0);
+
+            // assert that the pre-existing file is deleted
+            expect(fs.existsSync(preExistingFilePath)).toBeFalsy;
+        });
+
+        it(frontend + ': Should not clear the output directory before generation if cleanGeneratedModelsDirectory is false', async () => {
+            // mock the input and output file structure
+            const schemaFilePath = path.join(MOCK_BACKEND_DIRECTORY, 'api', MOCK_PROJECT_NAME);
+            const outputDirectory = path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend]);
+            const preExistingFileName = 'preexisting.txt';
+            const preExistingFilePath = path.join(outputDirectory, preExistingFileName);
+            const mockedFiles = {};
+            mockedFiles[schemaFilePath] = {
+                'schema.graphql': ' type SimpleModel { id: ID! status: String } '
+            };
+            mockedFiles[outputDirectory] = {};
+            mockedFiles[outputDirectory][preExistingFileName] = 'A pre-existing file to be cleared';
+            mock(mockedFiles);
```

Comment:
Could you rebase this PR and change the import to `mockFs` as the previous review?